.circleci/config.yml

```diff
@@ -240,120 +240,91 @@ jobs:
+              echo "No active environment for $DEV_BUILD_BRANCH found, attempt to activate one."
+              /home/circleci/.platformsh/bin/platform environment:activate -p $PLATFORM_PROJECT_ID -e $DEV_BUILD_BRANCH -y
+            fi
 
-              # Fetch database from our defined endpoint
-              git clone --depth=1 ${DB_SOURCE_URL} /home/circleci/dbs
-              # Restore to db service
-              cd /home/circleci/project/web
-              gzip -dc /home/circleci/dbs/nidirect--ui-test-sanitised.sql.gz | ../vendor/bin/drush sqlc
-              # Remove downloaded db file.
-              sudo rm -rf /home/circleci/dbs
-              # Clear caches/container rebuild for this environment.
-              ../vendor/bin/drush cr
-      - run:
-            name: Copy files into webroot
-            command: |
-              # Copy our build into position (./ suffix ensures hidden files are copied too).
-              sudo rsync -avq /home/circleci/project/. /var/www/html
-      - run:
-            name: UI tests with Nightwatch.js
-            command: yarn --cwd=/var/www/html/web/core test:nightwatch --skiptags core
-  # Debug job(s) - useful for working with circleci local which doesn't support caches or workspace persistence.
-  debug:
+  # Separate task to allow us to sync data on PSH environments, without pauses in other jobs.
+  sync_data:
     docker:
       - image: circleci/php:7.3.14-apache-browsers
     steps:
-      # Needed for circle ci local for some odd reason.
-      - github_hosts_workaround
       - checkout_code
       - install_php_os_extensions
-      - composer_tasks__no_cache
       - run:
-          name: PHPCS static analysis
-          command: /home/circleci/project/phpcs.sh /home/circleci/project "/home/circleci/project/web/modules/origins /home/circleci/project/web/modules/custom /home/circleci/project/web/modules/migrate /home/circleci/project/web/themes/custom"
+          name: Add extra OS and PHP extensions/config
+          command: |
+            sudo docker-php-ext-install pcntl posix
       - run:
-          name: Deprecated code check
+          name: Keyscan Platform.sh region hostnames for password-less access
           command: |
-            cd /home/circleci/project
-            vendor/bin/drupal-check /home/circleci/project/web/modules/custom /home/circleci/project/web/modules/origins
+            mkdir -p ~/.ssh
+            ssh-keyscan -H ssh.$PLATFORM_REGION >> ~/.ssh/known_hosts
+      - run:
+          name: Install the Platform.sh CLI tool
+          command: |
+            curl -sS https://platform.sh/cli/installer | php
+      - run:
+          name: Trigger a data sync from master environment to nightly edge build.
+          command: |
+            /home/circleci/.platformsh/bin/platform sync data -p $PLATFORM_PROJECT_ID -e $DEV_BUILD_BRANCH -y
```

Comment:
Things like $DEV_BUILD_BRANCH are stored in Circle CI's environment variable store. You can't see the value once created, so to update you basically have to remove + recreate. I think you can see the value if you run a job with SSH enabled, then SSH into the running container and use the `env | sort` trick to show you the values stored in the OS.